src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -20,20 +20,21 @@ public partial class Ping
         private const int MaxIpHeaderLengthInBytes = 60;
         private static bool SendIpHeader => OperatingSystem.IsMacOS();
         private static bool NeedsConnect => OperatingSystem.IsLinux();
+        private static bool IsOSXLike = OperatingSystem.IsMacOS() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS();
         [ThreadStatic]
         private static Random? t_idGenerator;
 
         private PingReply SendPingCore(IPAddress address, byte[] buffer, int timeout, PingOptions? options)
         {
-            PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) ?
+            PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) || IsOSXLike ?
```

Comment:
It'd be much better to have macOS/iOS/tvOS flavour of this assembly with implementation which does only `return SendIcmpEchoRequestOverRawSocket(address, buffer, timeout, options)` to be also size optimal with no ping utility dependency.